Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -431,13 +453,37 @@ void Connection::setDidCloseOnConnectionWorkQueueCallback(DidCloseOnConnectionWo
     m_didCloseOnConnectionWorkQueueCallback = callback;    
 }
 
+bool Connection::open(Client& client)
+{
+    ASSERT(!m_client);
+    if (!platformPrepareForOpen())
+        return false;
+    m_client = &client;
+    m_syncState = &SyncMessageState::get(RunLoop::main());
+    platformOpen();
+    return true;
+}
+
+#if !USE(UNIX_DOMAIN_SOCKETS)
+bool Connection::platformPrepareForOpen()
+{
+    return true;
+}
+#endif
+
 void Connection::invalidate()
 {
     ASSERT(RunLoop::isMain());
     m_isValid = false;
     if (!m_client)
         return;
     m_client = nullptr;
+    SyncMessageState* syncState = nullptr;
+    {
+        Locker locker { m_incomingMessagesLock };
+        syncState = std::exchange(m_syncState, nullptr);
+    }
+    syncState->release();
```

Comment:
I feel sort of similar here, with the use of a manual `release`. I think we only need to rename this to `deref`, and then `RefPtr<>` will work.

It's only a single callsite (and is unlikely to ever be more than that), but we have a project-wide convention of avoiding manual refcounting and relying on the auto wrapper objects, so I don't see a reason to deviate here.

Reply:
But it's not a RefPtr if it doesn't really support RefPtr. This is why I would not want to shoehorn it using it.

It's relatively simple use in here: the destructor asserts that the code in question has run.

I added a bit of convolution but can fix it up in further patch when moving SyncMessageState to its own file to simplify the implementations.